timesketch/frontend-ng/src/components/ShareCard.vue

```diff
@@ -0,0 +1,237 @@
+      <br />
+      General access
+      <v-select
+        hide-details
+        single-line
+        :items="items"
+        v-model="isPublic"
+        :label="generalAccess"
+        :prepend-icon="accessIcon"
+        style="width: 150px"
+        @change="setPublicAccess()"
+      ></v-select>
+      <v-card-actions>
+        <v-spacer></v-spacer>
+        <v-btn text @click="$emit('close-dialog')">Cancel</v-btn>
+        <v-btn color="primary" depressed @click="grantAccess()">Done</v-btn>
+      </v-card-actions>
+    </div>
+  </v-card>
+</template>
+
+<script>
+import ApiClient from '../utils/RestApiClient'
+
+export default {
+  data() {
+    return {
+      items: [
+        {
+          text: 'Public',
+          value: true,
+        },
+        {
+          text: 'Restricted',
+          value: false,
+        },
+      ],
+      usersAndGroupsToAdd: [],
+      usersToRemove: [],
+      groupsToRemove: [],
+      systemUsers: [],
+      systemGroups: [],
+      isPublic: false,
+      isLoading: false,
+    }
+  },
+  computed: {
+    sketch() {
+      return this.$store.state.sketch
+    },
+    meta() {
+      return this.$store.state.meta
+    },
+    generalAccess() {
+      if (this.isPublic) {
+        return 'Public'
+      }
+      return 'Restricted'
+    },
+    accessIcon() {
+      if (this.isPublic) {
+        return 'mdi-earth'
+      }
+      return 'mdi-lock-outline'
+    },
+    usersAndGroups() {
+      return this.systemUsers.concat(this.systemGroups)
+    },
+  },
+  methods: {
+    grantAccess: function () {
+      if (!this.usersAndGroupsToAdd.length) {
+        this.$emit('close-dialog')
+        return
+      }
+      this.isLoading = true
+      let usersToAdd = this.usersAndGroupsToAdd
+        .filter((user) => user.type === 'user')
+        .map((userObject) => userObject.value)
+      let groupsToAdd = this.usersAndGroupsToAdd
+        .filter((group) => group.type === 'group')
+        .map((groupObject) => groupObject.value)
+      ApiClient.grantSketchAccess(this.sketch.id, usersToAdd, groupsToAdd)
+        .then((response) => {
+          this.$store.dispatch('updateSketch', this.sketch.id).then(() => {
+            this.isLoading = false
+            this.usersAndGroupsToAdd = []
+            this.$emit('close-dialog')
+            this.successSnackBar('Updated sharing')
+          })
+        })
+        .catch((e) => {
+          this.errorSnackBar('Failed to share sketch ' + e)
+        })
+    },
+    revokeAccess: function (userName, groupName) {
+      this.isLoading = true
+      let userToRemove = [userName]
+      let groupToRemove = [groupName]
+      ApiClient.revokeSketchAccess(this.sketch.id, userToRemove, groupToRemove)
+        .then((response) => {
+          this.$store.dispatch('updateSketch', this.sketch.id).then(() => {
+            this.isLoading = false
+          })
+        })
+        .catch((e) => {
+          this.errorSnackBar('Failed to remove access ' + e)
```

Comment:
I suggest we also add the sketchid, user and or group to that error message so that it is clear to the user what was attempted

Reply:
The sketchID is clear to the user already (they are in the sketch). And the user/group should be clear as well because the can only remove one at a time. The error message returned should be informative for the end user.

I opt to keep it like this.